repo/fsrepo/fsrepo.go

```diff
@@ -572,25 +573,34 @@ var _ io.Closer = &FSRepo{}
 var _ repo.Repo = &FSRepo{}
 
 // IsInitialized returns true if the repo is initialized at provided |path|.
-func IsInitialized(path string) bool {
+func IsInitialized(path string, conf *config.Config) bool {
 	// packageLock is held to ensure that another caller doesn't attempt to
 	// Init or Remove the repo while this call is in progress.
 	packageLock.Lock()
 	defer packageLock.Unlock()
 
-	return isInitializedUnsynced(path)
+	return isInitializedUnsynced(path, conf)
 }
 
 // private methods below this point. NB: packageLock must held by caller.
 
 // isInitializedUnsynced reports whether the repo is initialized. Caller must
 // hold the packageLock.
-func isInitializedUnsynced(repoPath string) bool {
+func isInitializedUnsynced(repoPath string, conf *config.Config) bool {
 	if !configIsInitialized(repoPath) {
 		return false
 	}
 
-	if !util.FileExists(filepath.Join(repoPath, leveldbDirectory)) {
+	leveldbPath := path.Join(repoPath, leveldbDirectory)
+	if conf != nil {
+		dsPath := conf.Datastore.Path
+		if len(dsPath) != 0 {
+			leveldbPath = path.Join(dsPath, leveldbDirectory)
+		}
+
+	}
+
+	if !util.FileExists(leveldbPath) {
```

Comment:
Should there also be a check for the `flatfs` directory during all this initialization?